src/cromshell/utilities/cromshellconfig.py

```diff
@@ -0,0 +1,130 @@
+    "=includeKey=id&includeKey=executionStatus&includeKey=backendStatus&includeKey"
+    "=status&includeKey=callRoot&expandSubWorkflows=true&includeKey"
+    "=subWorkflowMetadata&includeKey=subWorkflowId"
+)
+api_string = "/api/workflows/v1/"
+
+
+def override_slim_metadata_parameters(slim_metadata_param):
+    """Override Cromwell Slim Metadata Key From Command Line"""
+
+    global slim_metadata_parameters
+
+    if slim_metadata_param:
+        slim_metadata_parameters = (
+                slim_metadata_param
+                + "&includeKey=subWorkflowMetadata&includeKey=subWorkflowId"
+        )
+
+
+def resolve_cromwell_config_server_address(server_user=None, workflow_id=None):
+    """
+    Override Cromwell Server From Command Line or Environment or Submission file
+
+    This function allows users to override the cromwell server set in the
+    cromshell config file using the command line options or environment
+    variables. By default the server URL in the config file is used, however
+    if an URL is specified in the command line or environment then this
+    overrides the default. If a workflow Id is provided without a server URL
+    then the server associated with a workflow id in the submission file
+    overrides the default. If a workflow ID is provided but isn't found
+    in the submission file then the default server is used.
+    """
+
+    global cromwell_server
+    global submission_file
+
+    if server_user is None and workflow_id is None:
+        LOGGER.info(
+            "Workflow id and cromwell server not specified. Using default cromwell server"
+        )
+        LOGGER.info(f"Server: {cromwell_server}")
+    elif server_user:
+        cromwell_server = server_user
+        LOGGER.info(
+            "Cromwell server URL was overridden by command line argument"
+        )
+        LOGGER.info(f"Server: {cromwell_server}")
+    else:
+        with open(submission_file, 'r') as csv_file:
+            reader = csv.DictReader(csv_file, delimiter="\t")
+            for row in reader:
+                if row["RUN_ID"] == workflow_id:
+                    LOGGER.info("Found workflow id in submission file.")
+                    cromwell_server = row["CROMWELL_SERVER"]
+                    LOGGER.info(
+                        "Cromwell server set to matching workflow id in submission file."
+                    )
+                    LOGGER.info(f"WorkflowID: {workflow_id}")
+                    LOGGER.info(f"Server: {cromwell_server}")
+                    break
+
+
+
+
+def __get_config_dir():
+    """Get Path To Cromshell Hidden Directory"""
+
+    config_path = str(Path.home()) + "/.cromshell"
+    Path.mkdir(Path(config_path), exist_ok=True)
+    return config_path
+
+
+def __get_submission_file(config_directory):
+    """Get File Path To Cromshell Submission File"""
+
+    submission_file_path = f"{config_directory}/all.workflow.database.tsv"
+    if not Path.exists(Path(submission_file_path)):
```

Comment:
I think this is a little cleaner:
```suggestion
    if not Path(submission_file_path).exists():
```